### ISSUE TITLE
Possible bug with concurrent stored procedure calls in 0.2.x

### ISSUE THREAD

**Issue by davazp** [TimelordUK/node-sqlserver-v8#34]
Hi,

I have reported the following issue in *node-mssql*, but it seems to be a bug in *msnodesqlv8* itself:

patriksimek/node-mssql#414

The bug seems to be introduced in the 0.2.x releases. I do not have an example using msnodesqlv8 directly just yet, but I reference the issue here so you are aware of it.

Thanks

**Comment by TimelordUK** [comment]
thanks for letting me know.   Yes, looks like driver is trying to run two queries at the same time on the same statement, which is not good news.  It may take me a few days to track down, I will report back when I dig into it further.

**Comment by TimelordUK** [comment]
if you are in a position to do so, could you explain the code a little i.e. what is the stored procedure doing, is it a simple select ....

im **msnodesqlv8.driverMgr.js**, could you run your test with a trace on 

what I am passing in for queryId with your test running.  In theory, each queryId should be managed entirely independently to all others on their own statements.  My first guess is that I am not doing that correctly.  Any additional information would be much appreciated.

    function procedure_internal(queryId, procedure, params, callback) {

        function onProc(err, results, params) {
            if (callback) {
                callback(err, results, params);
            }
        }
        //noinspection JSUnresolvedFunction
        return native.callProcedure(queryId, procedure, params, onProc);
    }

**Comment by davazp** [comment]
@TimelordUK I tried with multiple procedures, and the only requirement was for it to raise an error.

For instance,

```sql
create procedure test 
as begin
	   RAISERROR ('error', 16, 1);
end
```

would make it. However, the bug is there even when the stored procedure does not exist at all. So the query needs to fail.

I'll do some testing in `msnodesqlv8.driverMgr.js` later today.  Thanks!

**Comment by TimelordUK** [comment]
so far I have tried to implement the same sort of test directly over msnodesqlv8

I have run many times on  a single connection and it seems to run fine

I of course am not saying therefore it is not msnodesqlv8, I just want to try and isolate the test case else I will have to run yours via mssql to reproduce it.

[96] {
  "error": {
    "sqlstate": "42000",
    "code": 50000
  }
}
[97] {
  "error": {
    "sqlstate": "42000",
    "code": 50000
  }
}
[98] {
  "error": {
    "sqlstate": "42000",
    "code": 50000
  }
}
[99] {
  "error": {
    "sqlstate": "42000",
    "code": 50000
  }
}

    private storedProcedureStress(procedureDef:StoredProcedureDef, params:any) : Promise<any> {
        let inst = this;
        return new Promise((resolve, reject) => {
            ASQ().runner(function*() {
                let connection = yield inst.sqlWrapper.open();
                let array:any = [];

                yield inst.createProcedureDef(procedureDef);
                let count = 100;
                for (let i=0; i<count; i++){
                    array.push(i);
                }

                let raised = 0;
                let promises = array.map((i:any)=> {
                    let command = connection.getCommand().procedure(procedureDef.name).params(params);
                    // test should raise an error, or just don't exist at all
                    return command.execute().catch((err: any) => {
                        console.log(`[${raised}] ${JSON.stringify(err, null, 2)}`);
                        ++raised;
                    });
                });

                Promise.all(promises).then( ()=> {
                    resolve();
                }).catch((e:any)=> {
                    reject(e);
                });
            });
        });
    }

**Comment by TimelordUK** [comment]
This issue should now be resolved in 0.2.16 available on npm. The statement was not terminated correctly after raiseerror.

**Comment by davazp** [comment]
It worked for me as well. Thanks.
